rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -393,10 +385,6 @@ rmw_create_publisher(
     context_impl,
     "unable to get rmw_context_impl_s",
     return nullptr);
-  if (!context_impl->session_is_valid()) {
-    RMW_SET_ERROR_MSG("zenoh session is invalid");
-    return nullptr;
-  }
```

Comment:
Why remove this?

Reply:
The `z_check` function has been changed to internal in https://github.com/eclipse-zenoh/zenoh-c/pull/605. The check itself is to check the gravestone health of the zenoh data. If we want to check whether the session is constructed successfully, we should use the return code during the creation.

Reply:
Let's say the session was constructed successfully but was closed later on. How would we check if the session is still open?

Reply:
There's a function `z_session_is_closed` and it's been used as the implementation of `rmw_context_impl_s::Data session_is_valid`. I think I can add it back to this.

Reply:
Addressed in https://github.com/ZettaScaleLabs/rmw_zenoh/commit/62ca8225256782d57ac376c826975459aea9b417